### ISSUE TITLE
Improve initrock robustness re db delete

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2361]
Under certain manual rockstor service modifications including by-hand delete of the /opt/rockstor directory we see a likely avoidable initrock script fragility:
```
Mar 10 17:14:17 rleap15-3 initrock[21233]: Traceback (most recent call last):
Mar 10 17:14:17 rleap15-3 initrock[21233]:   File "/opt/rockstor/bin/initrock", line 40, in <module>
Mar 10 17:14:17 rleap15-3 initrock[21233]:     sys.exit(scripts.initrock.main())
Mar 10 17:14:17 rleap15-3 initrock[21233]:   File "/opt/rockstor/src/rockstor/scripts/initrock.py", line 387, in main
Mar 10 17:14:17 rleap15-3 initrock[21233]:     shutil.rmtree("/var/lib/pgsql/data")
Mar 10 17:14:17 rleap15-3 initrock[21233]:   File "/usr/lib64/python2.7/shutil.py", line 270, in rmtree
Mar 10 17:14:17 rleap15-3 initrock[21233]:     rmtree(fullname, ignore_errors, onerror)
Mar 10 17:14:17 rleap15-3 initrock[21233]:   File "/usr/lib64/python2.7/shutil.py", line 275, in rmtree
Mar 10 17:14:17 rleap15-3 initrock[21233]:     onerror(os.remove, fullname, sys.exc_info())
Mar 10 17:14:17 rleap15-3 initrock[21233]:   File "/usr/lib64/python2.7/shutil.py", line 273, in rmtree
Mar 10 17:14:17 rleap15-3 initrock[21233]:     os.remove(fullname)
Mar 10 17:14:17 rleap15-3 initrock[21233]: OSError: [Errno 2] No such file or directory: '/var/lib/pgsql/data/pg_stat_tmp/global.tmp'
Mar 10 17:14:17 rleap15-3 systemd[1]: rockstor-pre.service: Main process exited, code=exited, status=1/FAILURE
Mar 10 17:14:17 rleap15-3 systemd[1]: rockstor-pre.service: Failed with result 'exit-code'.
Mar 10 17:14:17 rleap15-3 systemd[1]: Failed to start Tasks required prior to starting Rockstor.
```
With the suspicious line as follows:

> OSError: [Errno 2] No such file or directory: '/var/lib/pgsql/data/pg_stat_tmp/global.tmp'

We use shutil.rmtree("/var/lib/pgsql/data") which is wipe any prior database and the underlying os.remove fails us.

The core of the issue here is that we should talk to db itself to request it clears house, not go in and do this for ourselves at the file level. As such this issue is likely a duplicate of:

"initrock should drop rockstor specific databases rather than del dir" #2076

**Comment by phillxnet** [comment]
Further investigation of a work around reveals that our heavy handed approach to clearing another's (postgresql's) house has left that service in a poorly state:
```
Mar 10 17:25:11 rleap15-3 systemd[1]: Stopped PostgreSQL database server.
Mar 10 17:25:15 rleap15-3 systemd[1]: Starting PostgreSQL database server...
Mar 10 17:25:15 rleap15-3 postgresql-script[21360]: postgres: could not find the database system
Mar 10 17:25:15 rleap15-3 postgresql-script[21360]: Expected to find it in the directory "/var/lib/pgsql/data",
Mar 10 17:25:15 rleap15-3 postgresql-script[21360]: but could not open file "/var/lib/pgsql/data/global/pg_control": No such file or directory
Mar 10 17:25:15 rleap15-3 postgresql-script[21358]: pg_ctl: could not start server
Mar 10 17:25:15 rleap15-3 postgresql-script[21358]: Examine the log output.
Mar 10 17:25:15 rleap15-3 systemd[1]: postgresql.service: Control process exited, code=exited, status=1/FAILURE
Mar 10 17:25:15 rleap15-3 systemd[1]: postgresql.service: Failed with result 'exit-code'.
Mar 10 17:25:15 rleap15-3 systemd[1]: Failed to start PostgreSQL database server.
```
This has not previously been observed and may be a new sensitivity in Leap 15.3 postgresql.

**Comment by phillxnet** [comment]
A brute force fix for a system found in this state, incomplete delete of the prior db file structure, is:
```
rleap15-3:~ # systemctl stop postgresql
rleap15-3:~ # rm -rf /var/lib/pgsql/data/
rleap15-3:~ # systemctl start postgresql
```
Which causes the db to re-initiate a-new by completing the in-completed wipe that lead to our broken state (via initrock's inappropriate actions). We then stop and start all rockstor services:
```
systemctl stop rockstor-bootstrap rockstor rockstor-pre
systemctl start rockstor-bootstrap
```
The Rockstor instance should then start up as if freshly re-installed as this procedure was initiated by a by-hand uninstall and delete of the /opt/rockstor directory, followed by a rockstor package install, which in turn ended up half-removing the db file structure as part of it's draconian meddling of postgresql's realm.

See the prior linked issue for our way forward on this. I.e. asking postgresql nicely to do our bidding.

**Comment by phillxnet** [comment]
Closing as a duplicate of #2076 and as:
Fixed by #2420
